### PR TITLE
TP2000-1009

### PR DESCRIPTION
**Pull request by dougmills-DIT**
# TP2000-1009
Update ME67 and add a test to verify it handles deleted measures correctly

## Why
 * In production deleting a measure still triggers ME67 which fails because it cant find the measure .... becuase its deleted

## What
* Altered the query in ME67 to find the measure, if it cant find it - its fine, dont check. 
* Added test to verify (which also failerd before update)
* All rules passing ion workbasket that originally seeded the ticket 

## Checklist
- Requires migrations? No 
- Requires dependency updates? No 

Links to relevant material
See: [Description](https://uktrade.atlassian.net/browse/TP2000-1009)

### CODE REVIEW

File: measures/business_rules.py [hunk 1]
```diff
@@ -1246,46 +1245,48 @@ def validate(self, exclusion):
...
+                # e.g. if a date range is 1/1/2020 to 31/1/2020, in the database the upper will be stored as 1/2/2020
+                # which means we must use gt rather than gte here. See the tests for ME67 for clarity - they all work
+                # correctly and the queried dates are all exactly within the ranges, no days space so we can be
+                # confident this rule is behaving as expected.
+                matching_members_to_exclusion_period = matching_members_to_exclusion_period.filter(
                     Q(valid_between__endswith__isnull=True)
                     | Q(
                         valid_between__endswith__isnull=False,
-                        valid_between__endswith__gt=measure.valid_between.upper,
+                        valid_between__endswith__gt=measures.last().valid_between.upper,
```
Comment (a-gleeson) [inline measures/business_rules.py:1284]: as used twice should we set latest measure above for maintainability?

File: measures/business_rules.py [hunk 2]
```diff
@@ -1334,8 +1335,11 @@ def validate(self, association):
 
 
 class ME71(FootnoteApplicability):
-    """Footnotes with a footnote type for which the application type = "CN footnotes"
-    cannot be associated with TARIC codes (codes with pos. 9-10 different from 00).
+    """
+    Footnotes with a footnote type for which the application type = "CN
+    footnotes" cannot be associated with TARIC codes (codes with pos.
+
```
Comment (a-gleeson) [inline measures/business_rules.py:1341]: extra line?

**Review by a-gleeson — APPROVED** [review]
LTGM!

File: measures/business_rules.py [hunk 3]
```diff
@@ -1246,46 +1245,48 @@ def validate(self, exclusion):
...
-        measure = Measure.objects.approved_up_to_transaction(self.transaction).get(
+        measures = Measure.objects.approved_up_to_transaction(self.transaction).filter(
             sid=exclusion.modified_measure.sid,
         )
 
-        geo_area = measure.geographical_area
-        members = geo_area.members.approved_up_to_transaction(
-            self.transaction,
-        )
+        # If no measures are found, this mean the measure is deleted and we can skip the check
```
[the remark this answers is not in the document]
Comment (dougmills-DIT) [reply measures/business_rules.py:1252]: That's correct, or it never existed I suppose, however no point checking exclusions against a measure that does not exist either way

File: measures/business_rules.py (hunk 1 above)
Comment (dougmills-DIT) [reply measures/business_rules.py]: updated

File: measures/business_rules.py (hunk 2 above)
Comment (dougmills-DIT) [reply measures/business_rules.py:1344]: updated

Comment (dougmills-DIT) [reply measures/business_rules.py]: ok, this is black being an idiot, removed full stop which makes it appear less broken

**Review by a-gleeson — APPROVED** [review]

**Pull request merged**